### PR TITLE
`SharedDirectoryMoveRequest` and `SharedDirectoryMoveResponse`

### PR DESCRIPTION
**Pull request by ibeckermayer**
Adds the Rust and Go piping and logic for `SharedDirectoryMoveRequest` and `SharedDirectoryMoveResponse`.

Corresponds to https://github.com/gravitational/webapps/pull/1045

### CODE REVIEW

File: lib/srv/desktop/rdp/rdpclient/client.go [hunk 1]
```diff
@@ -740,6 +738,27 @@ func (c *Client) sharedDirectoryWriteRequest(req tdp.SharedDirectoryWriteRequest
...
+
+func (c *Client) sharedDirectoryMoveRequest(req tdp.SharedDirectoryMoveRequest) C.CGOErrCode {
+	if c.cfg.AllowDirectorySharing {
+		if err := c.cfg.Conn.OutputMessage(req); err != nil {
+			c.cfg.Log.Errorf("failed to send SharedDirectoryMoveRequest: %v", err)
+			return C.ErrCodeFailure
+		}
+		return C.ErrCodeSuccess
+	}
+	return C.ErrCodeFailure
```
Comment (jakule) [inline lib/srv/desktop/rdp/rdpclient/client.go:759]: nit: For me this looks cleaner:
```suggestion
	if !c.cfg.AllowDirectorySharing {
		return C.ErrCodeFailure		
	}

	if err := c.cfg.Conn.OutputMessage(req); err != nil {
		c.cfg.Log.Errorf("failed to send SharedDirectoryMoveRequest: %v", err)
		return C.ErrCodeFailure
	}
	return C.ErrCodeSuccess
```

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs [hunk 2]
```diff
@@ -718,13 +722,35 @@ impl Client {
...
                     );
                 }
-                FileSystemInformationClassLevel::FileFsSizeInformation
-                | FileSystemInformationClassLevel::FileFsFullSizeInformation
-                | FileSystemInformationClassLevel::FileFsDeviceInformation => {
-                    return Err(not_implemented_error(&format!(
-                        "support for ServerDriveQueryVolumeInformationRequest with fs_info_class_lvl = {:?} is not implemented",
-                        rdp_req.fs_info_class_lvl
-                    )));
+                FileSystemInformationClassLevel::FileFsFullSizeInformation => {
```
Comment (jakule) [inline lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs:725]: I'm not a rust expert, but this code looks very repetitive. The simplest thing to do would be probably to extract the buffer creation to a separate function and here just return `self.prep_query_vol_info_response()` as looks like all arguments are always the same.

**Review by jakule — COMMENTED** [review]
Can you add some tests? I don't see any in either Go or Rust code.

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs (hunk 2 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs:715]: Good call, couldn't do exactly what was suggested but I could simplify: c8a86d941367639a0b050f529dfafc2eba2c471a

File: lib/srv/desktop/rdp/rdpclient/client.go (hunk 1 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/client.go]: I have a bunch of these so I fixed them all 45e8058b3d481e0b25fe7b37ac099e7cb1347a3a

**Comment by ibeckermayer** [comment]
> Can you add some tests? I don't see any in either Go or Rust code.

I'm of the opinion that its still more efficient to test this section manually (against a reference Microsoft RDP implementation) than to figure out the right way to do automated tests, so I'm continuing to go with that until we've implemented all the core functionality (create/read/write/move/delete) and then will go back and add automated tests.

**Review by jakule — APPROVED** [review]

**Review by xacrimon — APPROVED** [review]

**Pull request merged**